src/components/Instance/Types/String/index.svelte

```diff
@@ -1,11 +1,48 @@
-<div>TODO</div>
-
 <script>
 
+    console.log($$props);
+  export let definition = {};
+  let minLength, maxLength, pattern;
+
+  ({ minLength, maxLength, pattern } = {...definition });
+
+  function validNumber(n) { 
+    if (isNaN(n) || n < 0 ) {
+        return "invalid";
+    } 
+    return "valid";
+  };
+
 </script>
 
 <style>
-div {
-  background-color: red;
-}
+  .invalid {
+    background-color: pink;
+  }
+  .valid {
+    background-color: white;
+  }
+  .default {
+    background-color: grey;
+  }
 </style>
+
+<div>
+
+  {#if minLength}
+    <div class="{validNumber(minLength)}" >minLength {minLength}</div>
+  {:else }
+    <div class="default" >minLength 0</div>
+  {/if}
+
+  {#if definition.maxLength}
```

Comment:
We better use `definition.X` or `X` uniformly.

Reply:
ok, yeah, I was simply seeing what was available, sorry I need to clean that up